.github/workflows/flatpak.yml

```diff
@@ -0,0 +1,113 @@
+name: Flatpak Packages
+on:
+  push:
+    branches:
+      - main
+      - "releases/**"
+  pull_request:
+    branches:
+      - main
+      - "releases/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  manifest:
+    name: "Generate Manifest"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          path: src
+          sparse-checkout: |
+            linux/flatpak
+            Cargo.lock
+      
+      - name: Checkout tools
+        uses: actions/checkout@v4
+        with:
+          repository: flatpak/flatpak-builder-tools
+          path: tools
+      
+      - name: Install pip tools
+        shell: bash
+        run: pip install requirements-parser PyYAML toml aiohttp yq python-dateutil
+
+      - name: Update Cargo dependencies
+        shell: bash
+        run: src/linux/flatpak/flatpak-update-crates.sh src/Cargo.lock
+
+      - name: Generate Appstream metainfo
+        shell: bash
+        working-directory: src/linux/flatpak
+        run: ./appstream-metainfo.sh > org.mozilla.vpn.metainfo.xml
+
+      - name: Update git reference
+        shell: bash
+        working-directory: src/linux/flatpak
+        run: |
+          yq -yi 'del(.modules[-1].sources[0].branch)' org.mozilla.vpn.yml
+          yq -yi ".modules[-1].sources[0].url = \"https://github.com/${{ github.repository }}\"" org.mozilla.vpn.yml
+          yq -yi ".modules[-1].sources[0].commit = \"${{ github.sha }}\"" org.mozilla.vpn.yml
+          cat org.mozilla.vpn.yml
+
+      - name: Upload manifest
+        uses: actions/upload-artifact@v4
+        with:
+          name: mozillavpn-manifest
+          path: src/linux/flatpak
+
+  linter:
+    name: "Linters"
+    runs-on: ubuntu-latest
+    needs:
+      - manifest
+    
+    steps:
+      - name: Download manifest
+        uses: actions/download-artifact@v4
+        with:
+          name: mozillavpn-manifest
+          path: manifest/
+      
+      - name: Install Flatpak Linting Tools
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install flatpak flatpak-builder -y
+          flatpak remote-add --user --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+          flatpak install --user flathub -y org.flatpak.Builder
+      
+      - name: Run Manifest Linters
+        shell: bash
+        run: flatpak run --command=flatpak-builder-lint org.flatpak.Builder manifest manifest/org.mozilla.vpn.yml
+
+      - name: Run Appstream Linters
+        shell: bash
+        run: flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream manifest/org.mozilla.vpn.metainfo.xml
+
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
```

Comment:
I'm a bit worried about using `ubuntu-latest`. In releng stuff, we usually pin it to a specific major version.
Not a blocker, unless if you think this could cause problems in the future.

Reply:
I don't anticipate this would cause problems. Both the `build` and `linter` jobs are downloading and running containerized environments to do the real work, so we're not really making any significant use of the host Ubuntu environment.